### PR TITLE
SB-1231: Implement `npm install`

### PR DESCRIPTION
**Pull request by sbespalov**
- `npm install` implemented
- `npm view` implemented
- `strongbox-npm-metadata` dependency added

### CODE REVIEW

File: strongbox-parent/pom.xml [hunk 1]
```diff
@@ -59,7 +59,8 @@
         <version.carlspring.commons.io>1.0-SNAPSHOT</version.carlspring.commons.io>
         <version.carlspring.commons.http>1.0-SNAPSHOT</version.carlspring.commons.http>
         <version.carlspring.logback.config>1.0-SNAPSHOT</version.carlspring.logback.config>
-
+        <version.carlspring.npm.metadata>1.0.0-SNAPSHOT</version.carlspring.npm.metadata>
```
Comment (carlspring) [inline strongbox-parent/pom.xml]: I think a version of `1.0` would be sufficient, as all our other modules are like that.

Comment (sbespalov) [reply strongbox-parent/pom.xml]: changed to `1.0-SNAPSHOT`

**Comment by sbespalov** [comment]
@carlspring this PR is ready

**Comment by carlspring** [comment]
Great! I will review it and merge it today!

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/config/NpmLayoutProviderConfig.java [hunk 2]
```diff
@@ -42,7 +43,12 @@
     @Bean
     public ObjectMapper npmJackasonMapper()
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/config/NpmLayoutProviderConfig.java]: There's actually a typo here -- it's supposed to be `npmJacksonMapper`, not `npmJackasonMapper`. (Too many `a`-s in `Jackson`).

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java [hunk 3]
```diff
@@ -82,45 +95,187 @@
...
+
+    @Inject
+    private NpmPackageSupplier npmPackageSupplier;
+    
+    @RequestMapping(path = { "{storageId}/{repositoryId}/npm" }, method = RequestMethod.GET)
     public ResponseEntity<String> greet()
     {
-        //TODO: find out what NPM expect to receive here
+        // TODO: find out what NPM expect to receive here
         return ResponseEntity.ok("");
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java]: I don't think we need such a method. This was introduces by Alex back in the day for testing purposes. I think we could just remove this all-together.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/config/NpmLayoutProviderConfig.java (hunk 2 above)
Comment (carlspring) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/config/NpmLayoutProviderConfig.java]: @sbespalov : I am merging this pull, but could you please fix the typo in a follow-up pull? Thanks! :)

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java (hunk 3 above)
Comment (sbespalov) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java]: this method called by NPM client, so it's not just for testing

Comment (carlspring) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java]: Interesting! Thanks for pointing this out!

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/config/NpmLayoutProviderConfig.java (hunk 2 above)
Comment (sbespalov) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/config/NpmLayoutProviderConfig.java]: https://github.com/strongbox/strongbox/pull/792

**Pull request merged**